tests/integration/ha/test_network_cut.py

```diff
@@ -0,0 +1,426 @@
+
+    logger.info("Restoring network...")
+    try:
+        ha_helpers.network_release(machine_name)
+    except CalledProcessError:  # in case it was already cleaned up
+        pass
+
+    await asyncio.sleep(RESTART_DELAY * 2)
+    await extra_secure_wait_for_idle(ops_test)
+
+    logger.info("Checking Dashboard access...")
+    assert await access_all_dashboards(ops_test, pytest.relation.id)
+
+
+@pytest.mark.group(1)
+@pytest.mark.abort_on_fail
+async def test_network_cut_application(ops_test: OpsTest, request):
+    """SIGKILLs leader process and checks recovery + re-election."""
+    logger.info("Cutting all units from network...")
+
+    machines = []
+    for unit in ops_test.model.applications[APP_NAME].units:
+        machine_name = await ha_helpers.get_unit_machine_name(ops_test, unit.name)
+        ha_helpers.cut_unit_network(machine_name)
+        machines.append(machine_name)
+
+    await asyncio.sleep(RESTART_DELAY * 2)
+    ha_helpers.restore_unit_network(machine_name)
+    await ops_test.model.wait_for_idle(apps=[APP_NAME], status="active", timeout=1000)
+
+    # Check all nodes but the old leader
+    logger.info("Checking Dashboard access for the rest of the nodes...")
+    assert not (await access_all_dashboards(ops_test, pytest.relation.id))
+
+    logger.info("Restoring network...")
+    for machine_name in machines:
+        try:
+            ha_helpers.restore_unit_network(machine_name)
+        except CalledProcessError:  # in case it was already cleaned up
+            pass
+
+    await asyncio.sleep(RESTART_DELAY * 2)
+    await extra_secure_wait_for_idle(ops_test)
+
+    logger.info("Checking Dashboard access...")
+    assert await access_all_dashboards(ops_test, pytest.relation.id)
+
+
+@pytest.mark.group(1)
+@pytest.mark.abort_on_fail
+async def test_network_throttle_application(ops_test: OpsTest, request):
+    """SIGKILLs leader process and checks recovery + re-election."""
+    logger.info("Cutting all units from network...")
+
+    machines = []
+    for unit in ops_test.model.applications[APP_NAME].units:
+        machine_name = await ha_helpers.get_unit_machine_name(ops_test, unit.name)
+        ha_helpers.network_throttle(machine_name)
+        machines.append(machine_name)
+
+    await asyncio.sleep(RESTART_DELAY * 2)
+    await ops_test.model.wait_for_idle(apps=[APP_NAME], status="active", timeout=1000)
+
+    # Check all nodes but the old leader
+    logger.info("Checking Dashboard access for the rest of the nodes...")
+    assert not (await access_all_dashboards(ops_test, pytest.relation.id))
+
+    logger.info("Restoring network...")
+    for machine_name in machines:
+        try:
+            ha_helpers.network_release(machine_name)
+        except CalledProcessError:  # in case it was already cleaned up
+            pass
+
+    await asyncio.sleep(RESTART_DELAY * 2)
+    await extra_secure_wait_for_idle(ops_test)
+
```

Comment:
lets check no IP changed

Reply:
Done.

Reply:
check missing

Reply:
As said in https://github.com/canonical/opensearch-dashboards-operator/pull/22#discussion_r1591129618

This is independent from the functionality of the charm/application.

This is LXD + Juju working together, isn't it...?
Anyway, adding the check

Reply:
If the IP address change does not happen or happens when not expected to, our tests are effectively not testing what we wanted them to test

Reply:
Added as requestd.